packages/itwin/breakdown-trees/src/Views/visibility/DataLink.ts

```diff
@@ -10,100 +10,86 @@ export class DataLink {
-    return rows;
+
+    return this.executeQuery(iModel, query);
   }
 
   public static async queryStoryRange(iModel: IModelConnection, storyId: string, clipAtSpaces: boolean): Promise<Range3d | undefined> {
+    const runQuery = async (query: string) => {
+      const rows = await this.executeQuery(iModel, query);
+
+      if (rows.length > 0 && rows[0].minX)
+        return new Range3d(rows[0].minX, rows[0].minY, rows[0].minZ, rows[0].maxX, rows[0].maxY, rows[0].maxZ);
+      return undefined;
+    };
+
     const queryGetStoryWithElementsProps = `select MIN(spel.minX) as minX, MIN(spel.minY) as minY, MIN(spel.minZ) as minZ, MAX(spel.maxX) as maxX, MAX(spel.maxY) as maxY, MAX(spel.maxZ) as maxZ
     from spatialcomposition.CompositeOverlapsSpatialElements ovr join biscore.spatialIndex spel on spel.ECinstanceId = ovr.targetecinstanceid
     where sourceECInstanceId in
     (select sp.ecinstanceid from buildingspatial.story s join buildingspatial.space sp on s.ecinstanceid=sp.composingelement.id where s.ecinstanceid=${storyId}
     union
     select s.ecinstanceid from buildingspatial.story s where s.ecinstanceid=${storyId})`;
 
-    let queryGetStoreyProps;
-    if (clipAtSpaces) {
-      queryGetStoreyProps = `select MIN(spel.minX) as minX, MIN(spel.minY) as minY, MIN(spel.minZ) as minZ, MAX(spel.maxX) as maxX, MAX(spel.maxY) as maxY, MAX(spel.maxZ) as maxZ
+    const queryGetStoreyProps = `select MIN(spel.minX) as minX, MIN(spel.minY) as minY, MIN(spel.minZ) as minZ, MAX(spel.maxX) as maxX, MAX(spel.maxY) as maxY, MAX(spel.maxZ) as maxZ
       from spatialcomposition.CompositeComposesSubComposites ovr join biscore.spatialIndex spel on spel.ECinstanceId = ovr.targetecinstanceid
       where sourceECInstanceId in
       (select sp.ecinstanceid from buildingspatial.story s join buildingspatial.space sp on s.ecinstanceid=sp.composingelement.id where s.ecinstanceid=${storyId}
       union
       select s.ecinstanceid from buildingspatial.story s where s.ecinstanceid=${storyId})`;
-      let rows = await this.executeQuery(iModel, queryGetStoreyProps);
-      if (rows.length > 0 && rows[0].minX)
-        return new Range3d(rows[0].minX, rows[0].minY, rows[0].minZ, rows[0].maxX, rows[0].maxY, rows[0].maxZ);
-      rows = await this.executeQuery(iModel, queryGetStoryWithElementsProps);
-      if (rows.length > 0 && rows[0].minX)
-        return new Range3d(rows[0].minX, rows[0].minY, rows[0].minZ, rows[0].maxX, rows[0].maxY, rows[0].maxZ);
-    } else {
-      queryGetStoreyProps = queryGetStoryWithElementsProps;
-      const rows = await this.executeQuery(iModel, queryGetStoreyProps);
-      if (rows.length > 0 && rows[0].minX)
-        return new Range3d(rows[0].minX, rows[0].minY, rows[0].minZ, rows[0].maxX, rows[0].maxY, rows[0].maxZ);
-    }
-    return undefined;
+
+    return clipAtSpaces ? runQuery(queryGetStoreyProps) ?? runQuery(queryGetStoryWithElementsProps) : runQuery(queryGetStoryWithElementsProps);
```

Comment:
> return clipAtSpaces ? runQuery(queryGetStoreyProps) ?? runQuery(queryGetStoryWithElementsProps) : runQuery(queryGetStoryWithElementsProps);

The last 2 method calls are identical, we should be able to combine them.
Maybe something like `return clipAtSpaces && runQuery(queryGetStoreyProps) || runQuery(queryGetStoryWithElementsProps);` would be even better?

Reply:
fixed